scvi/model/_destvi.py

```diff
@@ -0,0 +1,282 @@
+            sc_model.module.n_latent,
+            sc_model.module.n_layers,
+            **model_kwargs,
+        )
+
+    def get_proportions(
+        self, adata: AnnData = None, keep_noise: bool = False
+    ) -> np.ndarray:
+        """
+        Returns the estimated cell type proportion for the spatial data.
+
+        Shape is n_cells x n_labels OR n_cells x (n_labels + 1) if keep_noise.
+
+        Parameters
+        ----------
+        adata
+            the anndata to loop through (used only if amortization)
+        keep_noise
+            whether to account for the noise term as a standalone cell type in the proportion estimate.
+        """
+        if self.is_trained_ is False:
+            raise RuntimeError("Please train the model first.")
+
+        adata = self._validate_anndata(adata)
+
+        column_names = self.cell_type_mapping
+        if keep_noise:
+            column_names = np.append(column_names, "noise_term")
+
+        if self.module.amortization in ["both", "proportion"]:
+            stdl = self._make_data_loader(adata=adata)
+            prop_ = []
+            for tensors in stdl:
+                generative_inputs = self.module._get_generative_input(tensors, None)
+                prop_local = self.module.get_proportions(
+                    x=generative_inputs["x"], keep_noise=keep_noise
+                )
+                prop_ += [prop_local.cpu()]
+            data = np.array(torch.cat(prop_))
+        else:
+            logger.info(
+                "No amortization for proportions, ignoring adata and returning results for the full data"
+            )
+            data = self.module.get_proportions(keep_noise=keep_noise)
+
+        return pd.DataFrame(
+            data=data,
+            columns=column_names,
+            index=self.adata.obs.index,
+        )
+
+    def get_gamma(self, adata=None) -> np.ndarray:
+        """
+        Returns the estimated cell-type specific latent space for the spatial data.
+
+        Shape is n_cells x n_latent x n_labels.
+        """
+        if self.is_trained_ is False:
+            raise RuntimeError("Please train the model first.")
+
+        adata = self._validate_anndata(adata)
+
+        if self.module.amortization in ["both", "latent"]:
+            stdl = self._make_data_loader(adata=adata)
+            gamma_ = []
+            for tensors in stdl:
+                generative_inputs = self.module._get_generative_input(tensors, None)
+                gamma_local = self.module.get_gamma(x=generative_inputs["x"])
+                gamma_ += [gamma_local.cpu()]
+            data = np.array(torch.cat(gamma_, dim=-1))
+        else:
+            logger.info(
+                "No amortization for latent values, ignoring adata and returning results for the full data"
+            )
+            data = self.module.get_gamma()
+        return np.transpose(data, (2, 0, 1))
+
+    def get_scale_for_ct(
+        self,
+        adata: AnnData,
+        y: int,
```

Comment:
y shouldn't be an int, it should be `cell_type: str` and you use the cell type mapping to get the correct int automatically, same for stereoscope

Reply:
the module level function can take the int, however